app/src/main/java/com/rockthevote/grommet/ui/eventFlow/PartnerLoginViewModel.kt

```diff
@@ -0,0 +1,115 @@
+package com.rockthevote.grommet.ui.eventFlow
+
+import androidx.lifecycle.*
+import com.rockthevote.grommet.data.api.RockyService
+import com.rockthevote.grommet.data.db.dao.PartnerInfoDao
+import com.rockthevote.grommet.data.db.model.PartnerInfo
+import com.rockthevote.grommet.util.coroutines.DispatcherProvider
+import com.rockthevote.grommet.util.coroutines.DispatcherProviderImpl
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+/**
+ * Created by Mechanical Man on 5/16/20.
+ */
+class PartnerLoginViewModel(
+        dispatchers: DispatcherProvider = DispatcherProviderImpl(),
+        private val rockyService: RockyService,
+        private val partnerInfoDao: PartnerInfoDao
+) : ViewModel() {
+
+    val partnerInfoId: LiveData<Long> =
+            Transformations.map(partnerInfoDao.getCurrentPartnerInfo()) { result ->
+                result.partnerInfoId
+            }
+
+    private val _partnerLoginState = MutableLiveData<PartnerLoginState>(PartnerLoginState.Init)
+    val partnerLoginState: LiveData<PartnerLoginState> = _partnerLoginState
+
+    private val _effect = MutableLiveData<PartnerLoginState.Effect?>()
+    val effect: LiveData<PartnerLoginState.Effect?> = _effect
+
+    private val coroutineExceptionHandler = CoroutineExceptionHandler { _, throwable ->
+        // TODO Should we handle this another way? Is it needed?
+        Timber.e(throwable)
+    }
+
+    private val supervisorJob = SupervisorJob()
+
+    private val rockyRequestScope = CoroutineScope(dispatchers.io + coroutineExceptionHandler)
+
+    init {
+        viewModelScope.launch(dispatchers.io) {
+            updateState(PartnerLoginState.Init)
+        }
+    }
+
+    fun validatePartnerId(partnerId: Long) {
+        updateState(PartnerLoginState.Loading)
+
+        if (partnerId == partnerInfoId.value) {
+            // just continue on if the value is the same
+            updateState(PartnerLoginState.Init)
+            updateEffect(PartnerLoginState.Success)
+
+        } else {
+            rockyRequestScope.launch {
+                runCatching {
+                    rockyService.getPartnerName(partnerId.toString())
+                }.onSuccess {
+                    //TODO not sure if I should still be checking for the response success if using "runcatching" ?
```

Comment:
`onSuccess` occurs as long as an exception wasn't thrown, so we still need to verify the request's success. eg. a 500 error would still be a success as far as this is concerned, because it doesn't throw an exception